scene/gui/base_button.cpp

```diff
@@ -348,13 +353,45 @@ Ref<Shortcut> BaseButton::get_shortcut() const {
 	return shortcut;
 }
 
+void BaseButton::_shortcut_feedback_timeout() {
+	in_shortcut_feedback = false;
+	queue_redraw();
+}
+
 void BaseButton::shortcut_input(const Ref<InputEvent> &p_event) {
 	ERR_FAIL_COND(p_event.is_null());
 
-	if (!is_disabled() && is_visible_in_tree() && !p_event->is_echo() && shortcut.is_valid() && shortcut->matches_event(p_event)) {
-		status.shortcut_press = true;
-		on_action_event(p_event);
+	if (!is_disabled() && p_event->is_pressed() && is_visible_in_tree() && !p_event->is_echo() && shortcut.is_valid() && shortcut->matches_event(p_event)) {
+		if (toggle_mode) {
+			status.pressed = !status.pressed;
+
+			if (status.pressed) {
+				_unpress_group();
+				if (button_group.is_valid()) {
+					button_group->emit_signal(SNAME("pressed"), this);
+				}
+			}
+
+			_toggled(status.pressed);
+			_pressed();
+
+		} else {
+			_pressed();
+		}
+		queue_redraw();
 		accept_event();
+
+		if (shortcut_feedback) {
+			if (shortcut_feedback_timer == nullptr) {
+				shortcut_feedback_timer = memnew(Timer);
+				add_child(shortcut_feedback_timer);
+				shortcut_feedback_timer->set_wait_time(0.5);
```

Comment:
Sounds like this might be a bit long, but we can start with this and see what the user feedback is. I'm sure @Calinou will be happy to tweak this :P

Reply:
BTW, could also use `SceneTreeTimer` in theory?
Seems like the only place we use it currently in core is for Viewport tooltips.

Reply:
I prefer the `Timer` node for this myself because you can restart it more easily, so you "push it back" instead of creating another one.